src/encoded/audit/file.py

```diff
@@ -373,6 +373,29 @@ def audit_file_in_correct_bucket(value, system):
         )
 
 
+def audit_file_matching_md5sum(value, system):
+    '''
+    Files with md5 sums matching other files should be marked with a WARNING audit.
+    If the other files are listed as matching but in fact have different md5 sums,
+    the file should be flagged with an ERROR for incorrect metadata.
+    '''
+
+    checked_statuses = ['released', 'revoked', 'archived', 'in progress']
+    if 'matching_md5sum' not in value or value.get('status') not in checked_statuses:
+        return
+
+    for matching_file in value.get('matching_md5sum'):
+        if matching_file.get('md5sum') != value.get('md5sum'):
+            detail = 'File {} is listed as having a matching md5 sum '.format(matching_file['@id']) + \
+                     'as file {}, but the files have different md5 sums.'.format(value['@id'])
+            yield AuditFailure('Incorrect matching_md5sum', detail, level='ERROR')
+        else:
+            detail = 'The md5 sum of file {} '.format(value['@id']) + \
+                     'matches that of file {}.'.format(matching_file['@id'])
+            yield AuditFailure('Matching md5 sums', detail, level='WARNING')
```

Comment:
I'm wondering what will happen for multiple matching files. I understand we need a warning to flag this file having matching files. But if we do this in the loop and there are 10 matching files, are we going to get 10 warnings or 1. And do we want to get 10 or 1 by design? @jenjou and @gabdank

Reply:
My understanding was we wanted a separate warning for each file, so it's written to raise 10 warnings instead of collecting them into 1. But I'd like to double check with @gabdank this is what we want.

Reply:
Changed this section, please view the updated file.